app/src/test/java/io/intrepid/contest/screens/sendinvitations/selectcontacts/SelectContactsPresenterTest.java

```diff
@@ -37,4 +43,51 @@ public void onViewBoundShouldGoBackToPreviousScreenWhenAppDoesNotHavePermissions
         presenter.onViewBound();
         verify(mockView).goBackToPreviousScreen();
     }
+
+    @Test
+    public void onContactListUpdatedShouldUpdateContactListFilteringOutContactsWithoutPhoneOrEmail() {
+        ArrayList<Contact> initialContactList = getMockContactList();
+        presenter.onContactListUpdated(initialContactList);
+        verify(mockView).updateContactList(argThat(argument -> argument.size() == initialContactList.size() - 1));
+    }
+
+    private ArrayList<Contact> getMockContactList() {
+        final String EMPTY = "";
+        final String TEST_PHONE = "555-555-5555";
+        final String TEST_EMAIL = "email@test.com";
+        ArrayList<Contact> contactList = new ArrayList<>();
+
+        Contact noPhoneOrEmail = new Contact();
+        noPhoneOrEmail.setPhone(EMPTY);
+        noPhoneOrEmail.setEmail(EMPTY);
+        contactList.add(noPhoneOrEmail);
+
+        Contact onlyPhone = new Contact();
+        onlyPhone.setPhone(TEST_PHONE);
+        onlyPhone.setEmail(EMPTY);
+        contactList.add(onlyPhone);
+
+        Contact onlyEmail = new Contact();
+        onlyEmail.setPhone(EMPTY);
+        onlyEmail.setEmail(TEST_EMAIL);
+        contactList.add(onlyEmail);
+
+        Contact phoneAndEmail = new Contact();
+        phoneAndEmail.setPhone(TEST_PHONE);
+        phoneAndEmail.setEmail(TEST_EMAIL);
+        contactList.add(phoneAndEmail);
+
+        return contactList;
+    }
+
+    @Test
+    public void onQueryTextSubmitShouldAlwaysReturnTrue() {
+        assertEquals(true, presenter.onQueryTextSubmit(""));
```

Comment:
Yeah, I wasn't sure how to test it, but I just figured since this is the desired behavior I should cover it as well.